src/XPurchase/VirtoCommerce.ExperienceApiModule.XOrder/Schemas/OrderSchema.cs

```diff
@@ -130,5 +149,30 @@ private async Task<object> ResolveConnectionAsync(IMediator mediator, IResolveCo
 
             return result;
         }
+
+        private async Task CheckAuthAsync(IResolveFieldContext context, object resource)
+        {
+            var userId = context.GetArgument<string>("userId");
+
+            if (userId == null)
+            {
+                throw new ExecutionError($"argument {nameof(userId)} is null");
+            }
+
+            var signInManager = _signInManagerFactory();
+            var user = await signInManager.UserManager.FindByIdAsync(userId);
```

Comment:
you can inject Func<UserManager> instead of SignInManager and as a result, you don't need to register Func<SignInManager> in the DI

Reply:
`Func<SignInManager<ApplicationUser>>` signInManagerFactory) is registered in DI for this call
```
var userPrincipal = await signInManager.CreateUserPrincipalAsync(user);
```

Reply:
Registered factory in the platform